example/examples/style-diffing.react.js

```diff
@@ -17,61 +17,61 @@
 // LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 // THE SOFTWARE.
-'use strict';
 
-var assign = require('object-assign');
-var React = require('react');
-var r = require('r-dom');
-var Immutable = require('immutable');
-var MapGL = require('../../src/index.js');
-var window = require('global/window');
+import Immutable from 'immutable';
+import window from 'global/window';
+
+import React, {PropTypes, Component} from 'react';
+import autobind from 'autobind-decorator';
+import MapGL from '../../src/index.js';
 
 // San Francisco
-var location = require('./../data/cities.json')[0];
+import SF_FEATURE from './../data/feature-example-sf.json';
+import CITIES from './../data/cities.json';
+const location = CITIES[0];
 
-function buildStyle(opts) {
-  opts = assign({
-    fill: 'red',
-    stroke: 'blue'
-  }, opts);
+function buildStyle({fill = 'red', stroke = 'blue'}) {
   return Immutable.fromJS({
     version: 8,
     name: 'Example raster tile source',
     sources: {
       'my-geojson-polygon-source': {
         type: 'geojson',
-        data: require('./../data/feature-example-sf.json')
+        data: SF_FEATURE
       }
     },
     layers: [
       {
         id: 'geojson-polygon-fill',
         source: 'my-geojson-polygon-source',
         type: 'fill',
-        paint: {'fill-color': opts.fill, 'fill-opacity': 0.4},
+        paint: {'fill-color': fill, 'fill-opacity': 0.4},
         interactive: true
       }, {
         id: 'geojson-polygon-stroke',
         source: 'my-geojson-polygon-source',
         type: 'line',
-        paint: {'line-color': opts.stroke, 'line-width': 4},
+        paint: {'line-color': stroke, 'line-width': 4},
         interactive: false
       }
     ]
   });
 }
 
-var StyleDiffingExample = React.createClass({
+const PROP_TYPES = {
+  width: PropTypes.number.isRequired,
+  height: PropTypes.number.isRequired
+};
 
-  displayName: 'StyleDiffingExample',
+export default class StyleDiffingExample extends Component {
 
-  PropTypes: {
-    width: React.PropTypes.number.isRequired,
-    height: React.PropTypes.number.isRequired
-  },
+  static get propTypes() {
```

Comment:
Is there any benefit to `static get propTypes() {` over...

````js
StyleDiffingExample.propTypes = {...};
````
If not, the latter seems more inline with the React docs.